src/cmds/pbs_rsub.c

```diff
@@ -839,24 +848,33 @@ main(int argc, char *argv[], char *envp[])
 						host = pattr->value;
 
 					if (pattr->resource && strcmp(pattr->name, ATTR_rescavail) == 0 && strcmp(pattr->resource, "ncpus") == 0)
-						ncpus = pattr->value;
+						ncpus_str = pattr->value;
+				}
+
+				ncpus = strtol(ncpus_str, &endp, 0);
+				if (*endp != '\0') {
+					fprintf(stderr, "pbs_rsub: Attribute value error\n");
+					CS_close_app();
+					exit(2);
 				}
 
-				if (strcmp(host, *hostp) == 0 && atoi(ncpus) > 0){
+				if (strcmp(host, *hostp) == 0 && ncpus > 0) {
 					if (!execvnodes_str) {
 						execvnodes_str = malloc(BUF_SIZE);
 						if (execvnodes_str == NULL) {
 							fprintf(stderr, "pbs_rsub: Out of memory\n");
+							CS_close_app();
 							exit(2);
 						}
 						execvnodes_str_size = BUF_SIZE;
 
-						sprintf(execvnodes_str, "(%s:ncpus=%s)", bstat->name, ncpus);
+						snprintf(execvnodes_str, BUF_SIZE - 1, "(%s:ncpus=%d)", bstat->name, ncpus);
 					} else {
-						sprintf(tmp_str, "+(%s:ncpus=%s)", bstat->name, ncpus);
+						snprintf(tmp_str, BUF_SIZE - 1, "+(%s:ncpus=%d)", bstat->name, ncpus);
```

Comment:
You don't need the -1.  snprintf() will only print N-1 bytes to leave space for the null byte.

Reply:
Right, fixed.